src/sentry/buffer/redis.py

```diff
@@ -110,6 +112,47 @@ def _make_pending_key_from_key(self, key):
     def _make_lock_key(self, key):
         return 'l:%s' % (key, )
 
+    def _dump_values(self, values):
+        result = {}
+        for k, v in six.iteritems(values):
+            result[k] = self._dump_value(v)
+        return result
+
+    def _dump_value(self, value):
+        if isinstance(value, six.string_types):
```

Comment:
I'm not sure if there will be any ramification of this, but this is glossing over the difference between unicode and bytes and treating them the same, which doesn't translate in JSON, but did translate in pickle.

I don't think it'll actually cause a problem, but if we put bytes into the buffer, unicodes are going to come out the other side since json.load will convert them to unicode.

Reply:
we could be explicit but i dont think there should be any bytes which is why i did this